bumble/controller.py

```diff
@@ -25,8 +25,6 @@
 import struct
 from bumble.colors import color
 from bumble.core import (
-    BT_CENTRAL_ROLE,
-    BT_PERIPHERAL_ROLE,
     BT_LE_TRANSPORT,
```

Comment:
Should those be remove as well and import `PhysicalTransport` instead?

Reply:
My initial idea is to keep this PR simple, focused on the major and necessary changes, and have another PR to cleanup them. But it's fine to cleanup now if we agree with the new design.

Reply:
Sounds good.